src/app/routes/app-routes.tsx

```diff
@@ -82,7 +82,7 @@ function useAppRoutes() {
           <Route
             element={
               <>
-                <MainHeader hideBackButton />
+                <MainHeader isHomePage hideBackButton />
```

Comment:
What is it we do differently if its the homepage? Can we rename this prop such that it describes what, not when, it does it?

Reply:
Got it. Just thinking through this, and wondering how we might build this in a way that avoids the "messiness", if you will, of having to tell the header what page its on. 

As I see it, the difficulty we keep encountering is that our generic header component (that we're trying to build as a single component), needs to handle use-case specific edge cases (that end up being part of the code component). E.g. as you've described above, rules specific to certain routes, but with exceptions.

Ideally, our Header component doesn't ever know where it's used, nor needs to be told (with props like `isHomepage`).

I reckon the problem lies within
1) the assumption the component only has a few variants without exceptions
2) the drive to minimise the amount of of code with each header definition

The solution is to find ways to compose the component, and being more explicit in its definition. If we have to repeat more header definition code, that's not a big problem.

Imagining an API like this, maybe using compound component pattern

```tsx
<Header title="Generic header">
 <Header.Logo /> {/* remove this on page where its not needed etc */}
 <Header.BackButton />
</Header>
```

Maybe this is repeated many times, so we create a wrapper component to avoid duplication, but then in the unlock screen or homepage, we define the header manually with this pattern. This could avoid the `isHomepage` prop.

Not suggesting we tackle this here, but worth considering for future iterations.

Reply:
Good idea. We have a header layout - `Header` that I am using to create a few use specific headers:
- `MainHeader`
- `PageHeader`
- `PopupHeader`

I think that to make this more elegant I can add two new header flavours:
- `HomeHeader`
- `UnlockHeader`

Then it will be more explicit as to what headers we use and those headers will compose the UI they need without using props. How does that sound?

Reply:
I've refactored them in [this commit](840e6758b64d17bc5ea3775d560e6ab12b67356a)  and I think it's a good improvement. Thanks!